data/binding/binding_test.go

```diff
@@ -29,8 +29,6 @@ func TestBase_AddListener(t *testing.T) {
 	})
 	data.AddListener(fn)
 	assert.Equal(t, 1, syncMapLen(&data.listeners))
-
```

Comment:
I think a new unit test where binding APIs are invoked in a goroutine could be worthwhile. The test when run normally just confirms the expected behavior, but when run with `go test -race` we can also prove that those APIs don't have any races.

Reply:
Good point. This is likely true for more parts of the API. We should run more tests with race instrumentation enabled (once we get there).

Reply:
I agree. I would prefer to do this as a follow-up PR. The intent of this work was to show that things do work exactly as before

Reply:
I have added an item to the TODO list on the parent ticket to come back to this.